minecraft/networking/packets.py

```diff
@@ -727,6 +727,399 @@ def __str__(self):
+        pass
+
+    class Type_Egg(Type):
+        pass
+
+    class Type_FireBall(Type):
+        pass
+
+    class Type_FireCharge(Type):
+        pass
+
+    class Type_Enderpearl(Type):
+        pass
+
+    class Type_Wither_Skull(Type):
+        pass
+
+    class Type_Shulker_Bullet(Type):
+        pass
+
+    class Type_Llama_spit(Type):
+        pass
+
+    class Type_Falling_Objects(Type):
+        pass
+
+    class Type_Item_frames(Type):
+        pass
+
+    class Type_Eye_of_Ender(Type):
+        pass
+
+    class Type_Potion(Type):
+        pass
+
+    class Type_Exp_Bottle(Type):
+        pass
+
+    class Type_Firework_Rocket(Type):
+        pass
+
+    class Type_Leash_Knot(Type):
+        pass
+
+    class Type_ArmorStand(Type):
+        pass
+
+    class Type_Evocation_Fangs(Type):
+        pass
+
+    class Type_Fishing_Hook(Type):
+        pass
+
+    class Type_Spectral_Arrow(Type):
+        pass
+
+    class Type_Dragon_Fireball(Type):
+        pass
+
+    def read(self, file_object):
+        self.entity_id = VarInt.read(file_object)
+        if self._context.protocol_version >= 49:
+            self.objectUUID = UUID.read(file_object)
+        type_id = Byte.read(file_object)
+        self.type = ClientSpawnObject.Type.type_from_id(type_id)
+
+        if self._context.protocol_version >= 100:
+            self.x = Double.read(file_object)
+            self.y = Double.read(file_object)
+            self.z = Double.read(file_object)
+        else:
+            self.x = Integer.read(file_object)
+            self.y = Integer.read(file_object)
+            self.z = Integer.read(file_object)
+
+        self.pitch = UnsignedByte.read(file_object)
+        self.yaw = UnsignedByte.read(file_object)
+        self.data = Integer.read(file_object)
+
+        if self._context.protocol_version < 49:
+            if self.data > 0:
+                self.velocity_x = Short.read(file_object)
+                self.velocity_y = Short.read(file_object)
+                self.velocity_z = Short.read(file_object)
+        else:
+            self.velocity_x = Short.read(file_object)
+            self.velocity_y = Short.read(file_object)
+            self.velocity_z = Short.read(file_object)
+
+    def write(self, socket, compression_threshold=None):
+        raise NotImplementedError
+
+
+class ClientBlockChange(Packet):
+    @staticmethod
+    def get_id(context):
+        return 0x0B if context.protocol_version >= 332 else \
+               0x0C if context.protocol_version >= 318 else \
+               0x0B if context.protocol_version >= 67 else \
+               0x24 if context.protocol_version >= 62 else \
+               0x23
+
+    packet_name = 'block change'
+
+    def read(self, file_object):
+        self.location = Position.read(file_object)
+        blockData = VarInt.read(file_object)
+        self.blockId = (blockData >> 4)
+        self.blockMeta = (blockData & 0xF)
+
+    def write(self, socket, compression_threshold=None):
```

Comment:
I would suggest implementing this so this is easier to test. See this commit for an example: https://github.com/ammaraskar/pyCraft/commit/d686b6487f8936fb6ed390386390632502b718b9